pkg/cfn/manager/nodegroup.go

```diff
@@ -76,7 +77,8 @@ func (c *StackCollection) createManagedNodeGroupTask(errorCh chan error, ng *api
 	name := c.makeNodeGroupStackName(ng.Name)
 
 	logger.Info("building managed nodegroup stack %q", name)
-	stack := builder.NewManagedNodeGroup(c.ec2API, c.spec, ng, builder.NewLaunchTemplateFetcher(c.ec2API), forceAddCNIPolicy, vpcImporter)
+	bootstrapper := nodebootstrap.NewManagedBootstrapper(c.spec, ng)
+	stack := builder.NewManagedNodeGroup(c.ec2API, c.spec, ng, builder.NewLaunchTemplateFetcher(c.ec2API), bootstrapper, forceAddCNIPolicy, vpcImporter)
```

Comment:
big nit: can this function get moved into a `managed_nodegroup.go` file? I find it weird that it lives in the `unmanaged` `nodegroup.go` file.

Reply:
This file contains a lot of stuff related to managed nodegroups. I think at this point `nodegroup` refers to both managed and unmanaged nodegroups.

Reply:
thats extremely confusing! We should definitely refactor it out